sdk/metric/internal/exemplar/reservoir_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exemplar
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Sat Jan 01 2000 00:00:00 GMT+0000.
+var staticTime = time.Unix(946684800, 0)
+
+type factory[N int64 | float64] func(requstedCap int) (r Reservoir[N], actualCap int)
+
+func ReservoirTest[N int64 | float64](f factory[N]) func(*testing.T) {
```

Comment:
This doesn't appear to be called anywhere?  Should we move this to the follow-up PR that implements the behavior?

Reply:
Having this in this PR allows the follow up PRs to be split. One to add the histogram implementation, one to add the drop, and one to add the fixed size implementation. I can move to one of those PRs if you think it better. Thoughts?

Reply:
Makes sense; thats fine.